src/main/java/com/amazon/jenkins/ec2fleet/CloudNanny.java

```diff
@@ -22,9 +25,14 @@
 
     private static final Logger LOGGER = Logger.getLogger(CloudNanny.class.getName());
 
+    private final ConcurrentMap<EC2FleetCloud, AtomicInteger> recurrenceCounters = new MapMaker()
+            .weakKeys() // the map should not hold onto fleet instances to allow deletion of fleets.
+            .concurrencyLevel(2)
```

Comment:
could be one as we have only one thread per jenkins to process.

Reply:
sure.